src/error.rs

```diff
@@ -0,0 +1,66 @@
+use std::error;
+use std::fmt::{self, Debug, Display};
+use std::io;
+use std::result;
+
+use serde::de;
+use serde::ser;
+
+use failure::format_err;
+use failure::Fail;
+
+pub struct Error {
```

Comment:
I'm not sure what this module is for.
It doesn't seem to be useful for anything.

Maybe we can just stick to using std::Error and derive our enum around it?
like https://github.com/arcnmx/serde-value/blob/master/src/de.rs#L81